lightningd/pay.c

```diff
@@ -42,6 +42,34 @@ struct sendpay_command {
 	struct command *cmd;
 };
 
+static bool string_to_payment_status(const char *status_str, enum wallet_payment_status *status)
```

Comment:
@ZmnSCPxj I understood your suggestions inside this function? or I'm introducing some mistake?

Reply:
looks ok, but no time for in-depth review now, please wait....